index.js

```diff
@@ -1,1144 +0,0 @@
-// Readline interface setup /////////////////////////////////////////////////////////////////////
-/////////////////////////////////////////////////////////////////////////////////////////////////
-
-const readline = require("readline");
-const readlineInterface = readline.createInterface(
-  process.stdin,
-  process.stdout
-);
-
-function ask(questionText) {
-  return new Promise((resolve, reject) => {
-    readlineInterface.question(questionText, resolve);
-  });
-}
-
-// Setting up character, room and item classes and player object ////////////////////////////////
-/////////////////////////////////////////////////////////////////////////////////////////////////
-
-let player = {
-  inventory: ["A Magnifying Glass"],
-  name: "",
-  location: [],
-  hasSpokenTo: [],
-  hasMoved: false,
-};
-
-let characterFactory = (name, location) => {
-  return { name, location };
-};
-
-let green = characterFactory("Mr. Green", "The Kitchen");
-
-let mustard = characterFactory("Colonel Mustard", "The Lounge");
-
-let scarlet = characterFactory("Miss Scarlet", "The Dining Room");
-
-let peacock = characterFactory("Mrs. Peacock", "The Lounge");
-
-let plum = characterFactory("Professor Plum", "The Library");
-
-let body = characterFactory("Mr. Body", "The Kitchen");
-
-let white = characterFactory("Mrs. White", "The Study");
-
-let cook = characterFactory("The Cook", "The Kitchen");
-
-let roomFactory = (name, desc, ...args) => {
-  inventory = args;
-  return { name, desc, inventory };
-};
-
-let kitchen = roomFactory(
-  "The Kitchen",
-  "\nYou see a beautiful kitchen with many appliances including a nice stove. You can see that the kitchen connects to the Dining Room, the Conservatory, the Library and the Study.",
-  "Frying Pan",
-  "Body",
-  "Stove"
```

Comment:
Passing these in as an array could be a better option rather than just passing the strings. With arrays you can use the pop and push methods to remove and add items into the room

Reply:
This comment goes for each of the rooms as well.

Reply:
The "args" on the room constructor takes multiple inputs (everything after the room description) and creates an array with them so they should already be in an array.

Reply:
`args` doesn't actually create an array. It creates an array-like object called an 'argument list' which doesn't quite have the same functionality as an array.

Reply:
Fixed